src/signals/incident/components/form/ContainerSelect/ContainerLayer/ContainerLayer.tsx

```diff
@@ -32,49 +34,86 @@ export const ContainerLayer: FunctionComponent<DataLayerProps> = ({ featureTypes
   );
 
   const getFeatureType = useCallback(
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-    feature => featureTypes.find(({ typeField, typeValue }) => feature.properties[typeField] === typeValue),
+    (feature: Feature) => featureTypes.find(({ typeField, typeValue }) => feature.properties[typeField] === typeValue),
     [featureTypes]
   );
 
-  const options: L.GeoJSONOptions<Record<string, string | undefined>> = useMemo(
+  type GeoJSONOptions = {
+    pointToLayer: (feature: Feature, latlng: LatLng) => L.Layer;
+  };
+  const options = useMemo<GeoJSONOptions>(
     () => ({
       pointToLayer: (feature, latlng: LatLng) => {
         const featureType = getFeatureType(feature);
         if (!featureType) return L.marker({ ...latlng, lat: 0, lng: 0 });
+        const selected = selection.some(
+          // Exclude from coverage; with the curent leaflet mock this can't be tested
+          /* istanbul ignore next*/({ id }) => id === feature.properties[featureType.idField]
+        );
 
-        return L.marker(latlng, {
+        const iconUrl = `data:image/svg+xml;base64,${btoa(
+          /* istanbul ignore next */ // Exclude from coverage; with the curent leaflet mock this can't be tested
+          selected ? featureType.icon.selectedIconSvg ?? '' : featureType.icon.iconSvg
+        )}`;
+
+        const marker = L.marker(latlng, {
           icon: L.icon({
             ...featureType.icon.options,
             className: featureType.label,
-            iconUrl: `data:image/svg+xml;base64,${btoa(featureType.icon.iconSvg)}`,
+            iconUrl,
           }),
           // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
           alt: feature.properties[featureType.idField],
         });
+
+        marker.on(
+          'click',
+          /* istanbul ignore next */ () => {
+            const { description, typeValue, idField }: Partial<FeatureType> = featureType;
+            const item = {
```

Comment:
Better to do something like 

`const item: Item = {`

and then cast the id field to string (turn off _non-nullable-type-assertion-style_?). That makes it more clear what part of the object needs to be casted.

It is also possible (maybe even better) to check if _id_ is defined.

Reply:
Good point! I'm not against turning of this rule, let's have a chat about that.